rustls/src/server/tls13.rs

```diff
@@ -715,6 +747,43 @@ mod client_hello {
         common.send_msg(c, true);
     }
 
+    fn emit_compressed_certificate_tls13(
+        transcript: &mut HandshakeHash,
+        common: &mut CommonState,
+        cert_chain: &[CertificateDer<'static>],
+        ocsp_response: Option<&[u8]>,
+        cert_compressor: &'static dyn CertCompressor,
+    ) {
+        let encoding =
+            CertificatePayloadTls13::new(cert_chain.iter(), ocsp_response).get_encoding();
+        let uncompressed_len = encoding.len() as u32;
+
+        let compressed = match cert_compressor.compress(encoding, CompressionLevel::Interactive) {
+            Ok(compressed) => PayloadU24(Payload::new(compressed)),
+            Err(_) => {
+                return emit_certificate_tls13(transcript, common, cert_chain, ocsp_response);
+            }
+        };
+
+        let body = CompressedCertificatePayload {
+            alg: cert_compressor.algorithm(),
+            uncompressed_len,
+            compressed,
+        };
```

Comment:
nit: I would be tempted to inline the `body` binding into the `Message` init on L778. It isn't ref'd anywhere.